controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentMetricsMaintainer.java

```diff
@@ -51,8 +53,10 @@ protected void maintain() {
                     }
                 }
                 catch (UncheckedIOException e) {
-                    log.log(Level.WARNING, "Failed talking to YAMAS: " + e.getMessage() +
-                            ", retrying in " + maintenanceInterval());
+                    if ( ! hasWarned) // produce only one warning per maintenance interval
+                        log.log(Level.WARNING, "Failed talking to YAMAS: " + Exceptions.toMessageString(e) +
+                                               ". Retrying in " + maintenanceInterval());
+                    hasWarned = true;
```

Comment:
Forgot to add this to the issue tracking this logging, but this warning was logged roughly once per 20 minutes on the controller yesterday, so it was already ~ once per run . This formatting is better, though.

Reply:
No, it is clear from the code it is not. And from this morning the log contains

{code}
[2017-10-26 09:02:38.190] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.190] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.190] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.191] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.191] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.191] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.191] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.192] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.192] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.192] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.192] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.193] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.193] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.193] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.194] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.194] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.194] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.194] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.195] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.195] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.195] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.196] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.196] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.196] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.196] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.197] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.197] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.197] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.197] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.198] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed talking to YAMAS: java.net.UnknownHostException: yamas.ops.yahoo.com, retrying in PT10M
[2017-10-26 09:02:38.198] WARNING : container        Container.com.yahoo.vespa.hosted.controller.maintenance.DeploymentMetricsMaintainer	Failed 
....
{code}

Reply:
I mean it was, in practice, once per run. Yesterday. Today: ouch! Change definitely warranted, then.